LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -138,7 +196,32 @@ List<Module> GetModuleList()
 
         public void CreateClassrooms(int classroomCount)
         {
-            throw new NotImplementedException();
+            string action = "Create Classroom";
+            int i = 0;
+            ReportHeader(actionName: action);
+
+            if (moduleIdList.Count == 0)
+            {
+                ReportError(missingResource: "Modules", failedAction: action);
+            }
+            else
+            {
+                while (i < classroomCount)
+                {
+                    Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
+                    Module module;
+                    moduleDictionary.TryGetValue(moduleId, out module);
```

Comment:
You have to check whether moduleDictionary.TryGetValue(moduleId, out module) returns true and them only add the module to the classroom

Reply:
DONE